### PR TITLE
feat: add support for archive storage class

### LINKED ISSUE #4: Add support for Archive storage class
Mirrors googleapis/google-cloud-java#6578 ; there is already an open PR at googleapis/google-cloud-java#6685 which should be re-created in this repo.

### PR DESCRIPTION
**Pull request by athakor**
Fixes #4

### CODE REVIEW

File: google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java [hunk 1]
```diff
@@ -41,22 +41,42 @@ public StorageClass apply(String constant) {
...
+  /**
+   * Regional storage class (deprecated). See: https://cloud.google.com/storage/docs/storage-classes
+   * for details
+   */
+  public static final StorageClass REGIONAL = type.createAndRegister("REGIONAL");
 
-  /** Durable Reduced Availability (deprecated) */
+  /**
+   * Multi-regional storage class (deprecated).
+   * See:https://cloud.google.com/storage/docs/storage-classes for details
```
Comment (kolea2) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java]: please add a space after the : (same below too)

**Comment by JesseLovelace** [comment]
We're avoiding the term "Deprecated" for these concepts currently and opting for "legacy". Please replace the (deprecated) notes with "This is supported as a legacy storage class and will be deprecated in the future"

File: google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java (hunk 1 above)
Comment (athakor) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java]: done

**Comment by athakor** [comment]
> We're avoiding the term "Deprecated" for these concepts currently and opting for "legacy". Please replace the (deprecated) notes with "This is supported as a legacy storage class and will be deprecated in the future"

done

File: google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java [hunk 2]
```diff
@@ -41,22 +41,43 @@ public StorageClass apply(String constant) {
...
 
-  /** Standard storage class. */
-  public static final StorageClass STANDARD = type.createAndRegister("STANDARD");
+  /**
+   * Archive storage class. See: https://cloud.google.com/storage/docs/storage-classes for details
+   */
+  public static final StorageClass ARCHIVE = type.createAndRegister("ARCHIVE");
+
+  /**
+   * Regional storage class.This is supported as a legacy storage class and will be deprecated in
```
Comment (kolea2) [inline google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java]: please add a space - `Regional storage class. This...` (same below)

Comment (athakor) [reply google-cloud-storage/src/main/java/com/google/cloud/storage/StorageClass.java]: done

**Comment by athakor** [comment]
@kolea2 @JesseLovelace PTAL

**Pull request merged**